packages/amplify-e2e-tests/src/__tests__/geo-populate.test.ts

```diff
@@ -0,0 +1,68 @@
+import {
+  createNewProjectDir,
+  deleteProject,
+  deleteProjectDir,
+  initJSProjectWithProfile,
+  addAuthWithDefault,
+  updateAuthAddUserGroups,
+  amplifyPushWithoutCodegen,
+  addGeofenceCollectionWithDefault,
+  populateGeofencesWithDefault,
+  getProjectMeta,
+  getGeofenceCollection
+} from 'amplify-e2e-core';
+import { existsSync } from 'fs';
+import path from 'path';
+import _ from 'lodash';
+
+describe('amplify geo populate', () => {
+  let projRoot: string;
+  beforeAll(async () => {
+    projRoot = await createNewProjectDir('geo-populate-test');
+    await initJSProjectWithProfile(projRoot, {});
+    await addAuthWithDefault(projRoot);
+    await updateAuthAddUserGroups(projRoot, ['admin']);
+    await addGeofenceCollectionWithDefault(projRoot, 'admin');
+  });
+
+  afterAll(async () => {
+    const metaFilePath = path.join(projRoot, 'amplify', '#current-cloud-backend', 'amplify-meta.json');
+    if (existsSync(metaFilePath)) {
+      await deleteProject(projRoot);
+    }
+    deleteProjectDir(projRoot);
+  });
+
+  it('should throw error if there is no geofence collection provisioned', async () => {
+    await expect(populateGeofencesWithDefault(projRoot)).rejects.toThrowError();
+  });
+
+  describe('GeoJSON uploading tests', () => {
+    beforeAll(async () => {
+      await amplifyPushWithoutCodegen(projRoot);
+      const meta = getProjectMeta(projRoot);
+      expect(meta.geo).toBeDefined();
+      const collectionId = _.findKey(meta.geo, ['service', 'GeofenceCollection']);
+      const collectionName = meta.geo[collectionId].output.Name;
+      const region = meta.geo[collectionId].output.Region;
+      const collection = await getGeofenceCollection(collectionName, region);
+      expect(collection.CollectionName).toBeDefined();
+    });
+    it('should upload a valid GeoJSON file with auto-generated root level ID to provisioned geofence collection', async () => {
+      await expect(populateGeofencesWithDefault(projRoot, { geoJSONFileName: 'valid-root-level-id.json' })).resolves.not.toThrow();
```

Comment:
for this test and the last one, we should make an SDK call to the resource and check that the data we expect is actually there.

Also, it looks like all of these tests are operating on the same project. In general it's bad to have shared resources between tests. Are there any potential resource conflicts / flakyness that this might cause?